pkg/types/workload/stats.go

```diff
@@ -0,0 +1,99 @@
+package workload
+
+import (
+	"log/slog"
+	"sync"
+
+	"github.com/els0r/goProbe/pkg/goDB/storage/gpfile"
+)
+
+// Workload stores all relevant parameters to load a block and execute a query on it
+type Workload struct {
+	workDirs []*gpfile.GPDir
+
+	stats      *Stats
+	statsFuncs StatsFuncs
+}
+
+// New creates a new workload
+func New(workDirs []*gpfile.GPDir) *Workload {
+	return &Workload{
+		workDirs: workDirs,
+		stats:    &Stats{Workloads: 1},
```

Comment:
The number of workloads can also be zero IIRC...

Reply:
Not sure I follow. If a workload is created, it must exist by definition.

Reply:
Ah, NVM, I misunderstood. The stats are updated in place, so the individual amount of "workloads" of a single `Stat` is of course 1. The plural name confused me...